service/mapper/transitions.go

```diff
@@ -97,52 +113,218 @@ func (t *Transitions) BootstrapState(s *State) error {
+	}
+	err = t.write.Header(s.height, header)
+	if err != nil {
+		return fmt.Errorf("could not index header: %w", err)
+	}
+	err = t.write.Guarantees(s.height, guarantees)
+	if err != nil {
+		return fmt.Errorf("could not index guarantees: %w", err)
+	}
+	err = t.write.Seals(s.height, seals)
+	if err != nil {
+		return fmt.Errorf("could not index seals: %w", err)
+	}
+
+	// Next, we try to retrieve the next commit until it becomes available,
+	// at which point all the data coming from the execution data should be
+	// available.
+	commit, err := t.chain.Commit(s.height)
+	if errors.Is(err, dps.ErrUnavailable) {
+		log.Debug().Msg("waiting for next state commitment")
+		time.Sleep(t.cfg.WaitInterval)
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("could not get commit: %w", err)
+	}
+	collections, err := t.chain.Collections(s.height)
+	if err != nil {
+		return fmt.Errorf("could not get collections: %w", err)
+	}
+	transactions, err := t.chain.Transactions(s.height)
+	if err != nil {
+		return fmt.Errorf("could not get transactions: %w", err)
+	}
+	results, err := t.chain.Results(s.height)
+	if err != nil {
+		return fmt.Errorf("could not get transaction results: %w", err)
+	}
+	events, err := t.chain.Events(s.height)
+	if err != nil {
+		return fmt.Errorf("could not get events: %w", err)
+	}
+
+	// Next, all we need to do is index the remaining data and we have fully
+	// processed indexing for this block height.
+	err = t.write.Commit(s.height, commit)
+	if err != nil {
+		return fmt.Errorf("could not index commit: %w", err)
+	}
+	err = t.write.Collections(s.height, collections)
+	if err != nil {
+		return fmt.Errorf("could not index collections: %w", err)
+	}
+	err = t.write.Transactions(s.height, transactions)
+	if err != nil {
+		return fmt.Errorf("could not index transactions: %w", err)
+	}
+	err = t.write.Results(results)
+	if err != nil {
+		return fmt.Errorf("could not index transaction results: %w", err)
+	}
+	err = t.write.Events(s.height, events)
+	if err != nil {
+		return fmt.Errorf("could not index events: %w", err)
+	}
+
+	// At this point, we need to forward the the `last` state commitment to
+	// `next`, so we know what the state commitment was at the last finalized
+	// block we processed. This will allow us to know when to stop when
+	// walking back through the forest to collect trie updates.
+	s.last = s.next
 
+	// Last but not least, we need to update `next` to point to the commit we
+	// have just retrieved for the new block height. This is the sentinel that
+	// tells us when we have collected enough trie updates for the forest to
+	// have reached the next finalized block.
+	s.next = commit
+
+	log.Info().Msg("indexed blockchain data for finalized block")
+
+	// After indexing the blockchain data, we can go back to updating the state
+	// tree until we find the commit of the finalized block. This will allow us
+	// to index the payloads then.
+	s.status = StatusUpdate
 	return nil
```

Comment:
```suggestion
	// After indexing the blockchain data, we can go back to updating the state
	// tree until we find the commit of the finalized block. This will allow us
	// to index the payloads then.
	s.status = StatusUpdate

	return nil
```